lib/elixir/lib/map_set.ex

```diff
@@ -375,6 +375,37 @@ defmodule MapSet do
     %{map_set | map: :sets.filter(pred, set)}
   end
 
+  @doc """
+  Splits the `MapSet` into two `MapSet`s according to the given function `fun`.
+
+  Splits the given `MapSet` into two maps by calling the provided `fun` which
+  receives each element in the `MapSet` as its only argument. Returns
+  a tuple with the first `MapSet` containing all the elements in `MapSet` for which
+  applying `fun` returned a truthy value, and a second `MapSet` with all the elements
+  for which applying `fun` returned a falsy value (`false` or `nil`).
+
+  ## Examples
+
+      iex> {while_true, while_false} = MapSet.split_with(MapSet.new([1, 2, 3, 4]), fn v -> rem(v, 2) == 0 end)
+      iex> while_true
+      #MapSet<[2, 4]>
+      iex> while_false
+      #MapSet<[1, 3]>
+
+      iex> {while_true, while_false} = MapSet.split_with(MapSet.new(), fn {_k, v} -> v > 50 end)
+      iex> while_true
+      #MapSet<[]>
+      iex> while_false
+      #MapSet<[]>
```

Comment:
```suggestion
      iex> {while_true, while_false} = MapSet.split_with(MapSet.new([1, 2, 3, 4]), fn v -> rem(v, 2) == 0 end)
      iex> while_true
      MapSet.new([2, 4])
      iex> while_false
      MapSet.new([1, 3])

      iex> {while_true, while_false} = MapSet.split_with(MapSet.new(), fn {_k, v} -> v > 50 end)
      iex> while_true
      MapSet.new([])
      iex> while_false
      MapSet.new([])
```

Reply:
Ohhh! The new inspect implementation! I realized I was working on a really old branch and didnt see the difference after I rebased! Very Nice!